hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupManager.java

```diff
@@ -514,24 +513,6 @@ public void addIncrementalBackupTableSet(Set<TableName> tables) throws IOExcepti
     systemTable.addIncrementalBackupTableSet(tables, backupInfo.getBackupRootDir());
   }
 
-  /**
-   * Saves list of WAL files after incremental backup operation. These files will be stored until
-   * TTL expiration and are used by Backup Log Cleaner plug-in to determine which WAL files can be
-   * safely purged.
-   */
-  public void recordWALFiles(List<String> files) throws IOException {
-    systemTable.addWALFiles(files, backupInfo.getBackupId(), backupInfo.getBackupRootDir());
```

Comment:
Can we get rid of the 'system' table altogether?

Reply:
Let me review and come back on the feasibility aspect.

Reply:
Did you figure anything @rda3mon ?

Reply:
@saintstack Sorry, I missed to reply on this one.

There is more cleanup possible, but not everything can be removed. Do you think It can be merged to `hbase:meta`?

Reply:
Created a ticket on this --> https://issues.apache.org/jira/browse/HBASE-26279